lib/src/main/java/graphql/nadel/NextgenEngine.kt

```diff
@@ -343,9 +343,16 @@ internal class NextgenEngine(
             serviceContext = executionContext.getContextForService(service).await(),
             serviceExecutionContext = serviceExecutionContext,
             hydrationDetails = executionHydrationDetails,
-            executableNormalizedField = transformedQuery,
+            // Prefer non __typename field first, otherwise we just get first
+            executableNormalizedField = topLevelFields
```

Comment:
So this `ServiceExecutionParameters.executableNormalizedField` actually needs to be deprecated and removed but we are keeping it for now for compatibility.

It's currently used to extract the cloud ID for routing.

The reason why we went from `ExecutableNormalizedField` to `List<ExecutableNormalizedField>` is because the root level hydration transform injects a `__typename` field.

```graphql
{
  batch_hydration__myIssues__myIssueKeys: myIssueKeys
  __typename__batch_hydration__myIssues: __typename
}
```

You could argue that we don't need a `__typename` for the operation types, but it's not a big deal to just keep it in.

Plus in the future we may end up grouping more root level fields into one execution, so this sort of lays the ground work for that.